### PR TITLE
NGO48 (EPSG:4273) to ETRS89 (EPSG:4258) triangulation-based transformation

### PR DESCRIPTION
**Pull request by himsve**
### NGO48 (EPSG:4273) to ETRS89 (EPSG:4258) triangulation-based transformation.

The transformation is based on the TIN-json file 'no_kv_ETRS89NO_NGO48_TIN.json'. The transformation in mentioned TIN-file has the direction EPSG:4258>EPSG:4273.

The transformation EPSG:4273>EPSG:4258 is already defined as EPSG:1653 at [https://epsg.io/1653](https://epsg.io/1653) as a Position Vector transformation (EPSG:9606). The triangulation-based NGO48>ETRS89 transformation is not yet registered with new EPSG code due to that triangulation-based transformation is not registered as method at epsg.io.

Preliminary the EPSG:4273>EPSG:4258 triangulation-based transformation is implemented as encapsulated proj string in  other_tranformation.

This pull request is related to #https://github.com/OSGeo/PROJ-data/pull/60

- [x]  Tests added. GIE test EPSG:4258>EPSG:27393.
- [x]  Added clear title that can be used to generate release notes
- [x] Resource file added to [https://github.com/OSGeo/PROJ-data](https://github.com/OSGeo/PROJ-data)
- [ ] EPSG code registration

### CODE REVIEW

File: data/Makefile.am [hunk 1]
```diff
@@ -124,6 +124,7 @@ EXTRA_DIST = proj.ini GL27 nad.lst nad27 nad83 \
 		tests/simple_model_wrap_west.tif \
 		tests/simple_model_projected.json \
 		tests/tinshift_crs_implicit.json \
+		tests/tinshift_simplified_etrs89_ngo48.json \
```
Comment (rouault) [inline data/Makefile.am]: I don't really think we need this file. The purpose of the existing tests/tinshift_ was to test the tinshift mechanism itself. Adding tinshift_simplified_etrs89_ngo48.json doesn't bring much.

File: data/sql/other_transformation_custom.sql [hunk 2]
```diff
@@ -1,5 +1,21 @@
 -- This file is hand generated.
 
+-- Norway triangulated files
+ 
+INSERT INTO other_transformation VALUES(
+    'PROJ','ETRS89NO_TO_NGO48','NGO 1948 to ETRS89 (2)',
```
Comment (rouault) [inline data/sql/other_transformation_custom.sql]:
```suggestion
    'PROJ','NGO48_TO_ETRS89NO','NGO 1948 to ETRS89 (2)',
```

File: data/sql/other_transformation_custom.sql [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
+    'PROJ','PROJString',
+    '+proj=pipeline ' ||
+        '+step +proj=axisswap +order=2,1 ' ||
+        '+step +proj=tinshift +file=no_kv_ETRS89NO_NGO48_TIN.json +inv ' ||
+        '+step +proj=axisswap +order=2,1',
+    'EPSG','4273',
+    'EPSG','4258',
+    0.1,
+    NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,0);
+INSERT INTO "usage" VALUES('PROJ','ETRS89NO_TO_NGO48_USAGE','other_transformation','PROJ','ETRS89NO_TO_NGO48','EPSG','1352','EPSG','1031');
```
Comment (rouault) [inline data/sql/other_transformation_custom.sql]:
```suggestion
INSERT INTO "usage" VALUES('PROJ','NGO48_TO_ETRS89NO_USAGE','other_transformation','PROJ','NGO48_TO_ETRS89NO','EPSG','1352','EPSG','1031');
```

File: test/gie/tinshift.gie [hunk 4]
```diff
@@ -47,4 +47,16 @@ accept      3210000.0000 6700000.0000   10.0
 expect      3210000.0000 6700000.0000   10.2886
 roundtrip   1
 
+# Test of Norwegian ETRS89 to NGO1948 transformation
```
Comment (rouault) [inline test/gie/tinshift.gie]: We probably don't need this

File: data/sql/other_transformation_custom.sql [hunk 5]
```diff
@@ -1,5 +1,21 @@
...
 
+-- Norway triangulated files
+ 
+INSERT INTO other_transformation VALUES(
+    'PROJ','NGO48_TO_ETRS89NO','NGO 1948 to ETRS89 (2)',
+    'Transformation based on a triangulated irregular network',
+    'PROJ','PROJString',
+    '+proj=pipeline ' ||
+        '+step +proj=axisswap +order=2,1 ' ||
+        '+step +proj=tinshift +file=no_kv_ETRS89NO_NGO48_TIN.json +inv ' ||
```
Comment (rouault) [inline data/sql/other_transformation_custom.sql:11]: the file shoul also be registered in grid_alternatives.sql similarly to the Finnish ones

**Comment by rouault** [comment]
@himsve Do you want to get this merged as it before EPSG comes with a code for the tinshift method  ?

**Comment by himsve** [comment]
Yes, please merged it. As I mentioned, this is just a temporary solution until EPSG handles TIN transformations. But it's works well. 👍

**Pull request merged**